src/content/docs/logs/ui-data/timestamp-support.mdx

```diff
@@ -0,0 +1,200 @@
+tags:
+  - Logs
+  - Timestamp
+  - Timestamps
+  - Forwarding Logs
+metaDescription: "How New Relic handles timestamps in logs."
+---
+
+## Introduction [#introduction]
+
+One of the most important attributes in a log event is the **timestamp**.  It is the exact moment in time that tells when the event happened and is critical for ordering events in chronological order as well as for searching logs in specific timeframes.  The most basic question all logs answer is **what happened, and when** it happened.  In most cases, the timestamp will appear somewhere in the beginning of the log event, but this is not always the case.  In some cases, it may appear much later in the log.  In logs that are in a JSON format they will usually contain some attribute that identifies the timestamp, like `timestamp` or `log_timestamp`, `time`, etc.  Some logs may not contain timestamps at all, and in these cases New Relic will always assign a timestamp automatically to the event at the time it is ingested into the system.  New Relic also indexes the timestamp of each log, since they are crucial for efficient querying.
+
+There are many, many different timestamp formats as you can see in some of these example logs below.  There is no single standard for the format of a timestamp.  They may or may not contain certain fields like the date, ms precision, or the timezone. 
+
+```
+time="2023-11-16T14:50:44Z" level=info msg="checking pixie-operator.v0.1.4"
+
+2023-11-16 16:12:49.245 ERROR Servlet.service() for servlet
+
+7:S 16 Nov 14:50:44.359 # Unable to connect to MASTER: Success
+
+102.28.1.1 - - [16/Nov/2023:09:50:44 -0500] "GET /index.html HTTP/1.1" 503 3000 "www.cdn-logs.com" "Mozilla/5.0 (Windows NT 4.0; en-US; rv:1.9.3.20) Gecko/1913-21-08 Firefox/37.0"
+
+I20231116 14:50:45.549868    40 exec.cc:63] Completed query: id=a1e9eac3-664a-4fab-8ece-ae6f19f23020
+
+{"bytes" : "10009528", "cacheStatus" : "1", "cliIP" : "172.16.103.13", "timestamp" : "1700146596212", "tlsOverheadTimeMSec" : "2"}
+```
+
+## How timestamps work for logs [#how_timestamps_work]
+
+**By default**, New Relic assigns the value of the log `timestamp` attribute (a reserved keyword) at the exact time when the log is ingested, i.e. ingest time.
+
+There are two options to override this default behavior and assign the `timestamp` value based on a `timestamp` contained in the log.
+1. If we receive a [JSON formatted log](#timestamps_in_json_logs) containing a `timestamp` attribute in one of our supported time formats, we will override the ingest timestamp value with the value of the timestamp from the json attribute.
+2. You can use [parsing rules](#timestamp_parsing_rules) at ingest to override the ingest `timestamp` value to the timestamp value that is contained in the log.  Using a parsing rule you can extract out our supported time formats with simple Grok expressions, or by using the `datetime` Grok type and a `pattern` to define the custom timestamp format in your log.
```

Comment:
@marcelschlapfer what does Grok type and Grok expressions mean in this context?

Reply:
A example of a Grok expression would be `%{NUMBER:timestamp:long}`, the grok type in this case is `long`.
Another example in the context of timestamps would be the expression `%{DATA:timestamp:datetime;MM/dd/yyyy HH:mm:ss.SSS}`, where the grok type is `datetime`.